client/src/extension.ts

```diff
@@ -35,24 +35,52 @@ const createClient = (
   );
 };
 
+let restartLanguageServer = async (context: ExtensionContext) => {
+  if (client) {
+    await client.stop();
+    client = undefined;
+  }
+
+  await activate(context);
+};
+
 /**
  * Set up handling for configuration changes.
- * @param {LanguageClient} client - The language client.
  */
-const handleConfigurationChanges = (client: LanguageClient) => {
-  workspace.onDidChangeConfiguration((event) => {
+const handleConfigurationChanges = (context: ExtensionContext) => {
+  workspace.onDidChangeConfiguration(async (event) => {
+    // Check if any configuration under 'slice' has changed
     if (event.affectsConfiguration("slice")) {
-      // Retrieve the updated configuration settings.
       const config = workspace.getConfiguration("slice");
-      const sourceDirectory = config.get<string>("sourceDirectory");
+      const enableLanguageServer = config.get<boolean>(
+        "languageServer.enabled"
+      );
       const referenceDirectories = config.get<string>("referenceDirectories");
 
-      // Send the "workspace/didChangeConfiguration" notification to the server with the updated settings.
-      client.sendNotification("workspace/didChangeConfiguration", {
-        settings: {
-          slice: { sourceDirectory, referenceDirectories },
-        },
-      });
+      // Send the updated configuration to the language server
+      if (client) {
+        client.sendNotification("workspace/didChangeConfiguration", {
+          settings: {
+            slice: { enableLanguageServer, referenceDirectories },
```

Comment:
I don't think we need to send this setting to the language server. Does it care if it's disabled?